cli/pcluster/config/param_types.py

```diff
@@ -1039,13 +1039,24 @@ class Section(object):
     def __init__(self, section_definition, pcluster_config, section_label=None):
         self.definition = section_definition
         self.key = section_definition.get("key")
-        self.label = section_label or self.definition.get("default_label", "")
+        self._label = section_label or self.definition.get("default_label", "")
         self.pcluster_config = pcluster_config
 
         # initialize section parameters with default values
         self.params = {}
         self._from_definition()
 
+    @property
+    def label(self):
+        """Get the section label."""
+        return self._label
+
+    @label.setter
+    def label(self, label):
+        """Set the section label. Marks the PclusterConfig parent for refreshing if called."""
+        self._label = label
+        self.pcluster_config.refresh()
```

Comment:
As per our conversation, the way the refresh() works is an internal detail of PclusterConfig, the main technical reason being to avoid unnecessary multiple refreshing cycles. In any case, the contract with the user remains the same, whether or not the actual refresh is performed in a lazy way. Of course let's re-open the conversation if needed.